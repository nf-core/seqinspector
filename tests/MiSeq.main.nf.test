nextflow_pipeline {

    name "Test Workflow main.nf on MiSeq data"
    script "../main.nf"
    tag "seqinspector"
    tag "PIPELINE"

    test("MiSeq data test") {

        when {
            config "./MiSeq.main.nf.test.config"
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/global_report/multiqc_data/multiqc_software_versions.txt"),
                    ).match()
                }
            )
        }
    }
}
