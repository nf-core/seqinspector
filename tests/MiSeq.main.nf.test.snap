{
    "MiSeq data test": {
        "content": [
            "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
            "multiqc_fastqc.txt:md5,7b1b7fd457b60404768045b148d4c0a8",
            "multiqc_general_stats.txt:md5,962713a1473a318f2cb29bb5290c4c8e",
            "multiqc_software_versions.txt:md5,49e3596d49ee49d967d3b6c363b486d5"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-10T08:57:05.870194"
    }
}